File: user-service/src/main/java/socialapp/userservice/repository/BlockRepository.java
package socialapp.userservice.repository;

import org.springframework.stereotype.Repository;
import socialapp.userservice.model.entity.Block;
import socialapp.userservice.repository.common.CommonRepository;

import java.util.Set;

@Repository
public interface BlockRepository extends CommonRepository<Block,Long> {
    @Override
    default Class<?> entityClass(){
        return Block.class;
    };

    Set<Block> findAllBySenderId(Long senderId);

}

File: kt-user-service/src/main/kotlin/socialapp/ktuserservice/repository/BlockRepository.kt
package socialapp.ktuserservice.repository

import org.springframework.stereotype.Repository
import socialapp.ktuserservice.model.entity.Block
import socialapp.ktuserservice.repository.common.CommonRepository

@Repository
interface BlockRepository: CommonRepository<Block,Long> {

    override fun entityClass(): Class<*> = Block::class.java

    fun findAllBySenderId(senderId: Long): Set<Block>
}